### PR TITLE
Automated cherry pick of #31275

### PR DESCRIPTION
**Pull request by dims**
Cherry pick of #31275 on release-1.3.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31279)

### CODE REVIEW

**Comment by Random-Liu** [comment]
LGTM. :) Thanks for fixing this. @dims

**Comment by dchen1107** [comment]
LGTM

**Comment by dchen1107** [comment]
cc/ @timstclair @fabioy our release czar for decision. This is the regression in 1.3 release, and the fix is very small and safe. I know we are in the process of cutting 1.3.6, could we also include this very fix? Thanks!

File: pkg/kubelet/dockertools/kube_docker_client.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (d *kubeDockerClient) StartContainer(id string) error {
 
 // Stopping an already stopped container will not cause an error in engine-api.
 func (d *kubeDockerClient) StopContainer(id string, timeout int) error {
-	ctx, cancel := d.getTimeoutContext()
+	ctx, cancel := d.getCustomTimeoutContext(time.Duration(timeout) * time.Second)
```
Comment (pracucci) [inline pkg/kubelet/dockertools/kube_docker_client.go]: Shouldn't we add a little delta to the input `timeout`?

**Pull request merged**